File: work/pmod_cpt1/pmod_cpt1.py
# Importer les fonctions nécessaires de gpio.py et time.py
from gpio import JA, JB, JC, gpio_set_direction, gpio_set_value
import time

# Définir le nombre de LEDs connectées et allumer leur direction sur "sortie"
num_leds = 8
for pin in JA:
    gpio_set_direction(pin, 1)

# Initialiser le compteur des LEDs allumées
led_count = 0

# Boucle infinie pour allumer progressivement les LEDs
while True:
    led_count += 1  # Incrémenter le compteur des LEDs allumées
    for i in range(num_leds):
        output_bit = (led_count >> i) & 1  # Récupérer le bit correspondant à l'indice i dans le compteur
        gpio_set_value(JA[i], output_bit)  # Allumer ou éteindre la LED correspondante en fonction du bit
    time.sleep(0.1)  # Attendre un petit moment avant de continuer

File: work/pmod_cpt1/pmod_cpt1.c
#include "/home/mi005558/pi/work/common/common.h"
#include "/home/mi005558/pi/work/gpio/gpio.h"

#define NUM_LEDS 8

int main()
{
    // Définition des broches utilisées pour les LEDS
    int led_pins[NUM_LEDS] = {8, 10, 9, 11, 19, 21, 20, 18};

    // Configuration des broches en sortie pour les LEDS
    for (int i = 0; i < NUM_LEDS; i++) {
        gpio_set_direction(led_pins[i], 1);
    }

    // Boucle principale du programme
    int binary_count = 0;
    while (1) {
        binary_count++;

        // Mise à jour de l'état des LEDS en fonction du compteur binaire
        for (int i = 0; i < NUM_LEDS; i++) {
            int output_bit = (binary_count >> i) & 1;
            gpio_set_value(led_pins[i], output_bit);
        }

        usleep(100000); // Pause de 100ms
    }

    return 0;
}

